jsdoc/valid-types error when namepath is numeric

## Expected behavior
Individual indexes in an Array can be documented. Note that jsdoc handles the below syntax without complaint:

<img width="333" height="257" alt="Image" src="https://github.com/user-attachments/assets/641e208d-d3a6-400b-a35d-fd5debfedfbe" />

## Actual behavior
```
  3:1  warning  Syntax error in namepath: '0'  jsdoc/valid-types
  4:1  warning  Syntax error in namepath: '1'  jsdoc/valid-types
```

## ESLint Config

<!--

What is the minimal config that reproduces the issue?

Please try to reproduce with only a single rule in your config (along
with the minimal number of options or settings that are needed to trigger
the error).

-->
```json
{
  "rules": {
    "jsdoc/valid-types": "error"
  }
}
```

## ESLint sample

<!-- What code triggers the error? -->

```js
/**
 * @typedef {Array} AnnotatedCharacter
 * @property {string} 0 Character data
 * @property {string[]} 1 Annotation hashses
 */
```

## Environment

- Node version: 20.19.5
- ESLint version 8.57.0
- `eslint-plugin-jsdoc` version: 62.5.0

--- a/docs/rules/valid-types.md
+++ b/docs/rules/valid-types.md
@@ -992,5 +992,12 @@ class Test {
 let SettingName = /** @type {const} */ ({
   THEME: `theme`,
 })
+
+/**
+ * @typedef {Array} AnnotatedCharacter
+ * @property {string} 0 Character data
+ * @property {string[]} 1 Annotation hashses
+ */
+// Settings: {"jsdoc":{"mode":"jsdoc"}}
 ````
 

--- a/src/rules/validTypes.js
+++ b/src/rules/validTypes.js
@@ -176,6 +176,15 @@ export default iterateJsdoc(({
             break;
           }
 
+          case 'prop':
+          case 'property': {
+            if (mode === 'jsdoc' && (/^\d+$/v).test(namepath)) {
+              handled = true;
+            }
+
+            break;
+          }
+
           case 'borrows': {
             const startChar = namepath.charAt(0);
             if ([

--- a/test/rules/assertions/validTypes.js
+++ b/test/rules/assertions/validTypes.js
@@ -2037,5 +2037,34 @@ export default /** @type {import('../index.js').TestCases} */ ({
         })
       `,
     },
+    {
+      code: `
+        /**
+         * @typedef {Array} AnnotatedCharacter
+         * @property {string} 0 Character data
+         * @property {string[]} 1 Annotation hashses
+         */
+      `,
+      settings: {
+        jsdoc: {
+          mode: 'jsdoc',
+        },
+      },
+    },
+    {
+      code: `
+        /**
+         * @typedef {Array} AnnotatedCharacter
+         * @prop {string} 0 Character data
+         * @prop {string[]} 1 Annotation hashses
+         */
+      `,
+      ignoreReadme: true,
+      settings: {
+        jsdoc: {
+          mode: 'jsdoc',
+        },
+      },
+    },
   ],
 });

